### PR TITLE
docs: Consolidate build rules

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: docs: Consolidate build rules
Additional Description:

This is the first of a few PRs to cleanup/consolidate the docs build

This one:

- updates `rules_pkg` to the newest version
- makes use of the newer `pkg_files` and `pkg_filegroup`

in follow up prs i intend to:

- consolidate the builder scripts (api, deps, extensions) into a single docs builder
- shift the sphinx_runner to the upstream package

Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: docs/BUILD [hunk 1]
```diff
@@ -171,14 +128,53 @@ genrule(
...
+    strip_prefix = "/docs",
+)
+
+pkg_files(
+    name = "sphinx_root",
+    srcs = glob(["root/**/*"]),
+    strip_prefix = "/docs/root",
+)
+
+pkg_files(
```
Comment (phlax) [inline docs/BUILD:148]: this doesnt seem to be included anywhere atm

**Review by wrowe — DISMISSED** [review]
This is fine, do you want to drop in a todo/comment about the unused/undeployed rule?

**Comment by phlax** [comment]
> do you want to drop in a todo/comment about the unused/undeployed rule?

yep, ill do it now

**Review by wrowe — APPROVED** [review]
Looks good. Merging.

**Pull request merged**